.gitignore

```diff
@@ -13,8 +13,6 @@ thumbs.db
 .bashrc
 .editorconfig
 
-# top-level go.mod is not meant to be checked in
-/go.mod
```

Comment:
Still _slightly_ on the fence if this could cause accidentally adding the go.mod (if the script failed at some point  etc), but I'll guess we'll see how it goes 🙈

Reply:
`hack/validate/no-module` added in this PR would catch those cases. Though it will need to be added to `hack/validate/default` to be run in CI?

Reply:
I excluded it from `default` because some users may want to be able to have the tooling function with a go.mod present but not committed. It's run in CI as there is a file-based glob. However, we could add it to `all` (the same step where vendoring is run).

Reply:
> It's run in CI as there is a file-based glob.

@neersighted I don't see it (neither in the current git HEAD, nor in the source tree during the state when this PR got merged.